### PR TITLE
Sharing services

### PR DESCRIPTION
**Pull request by dennisroberts71**
These are the changes to implement the app permission listing service.

### CODE REVIEW

File: services/apps/src/apps/routes/domain/permission.clj [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  (:use [common-swagger-api.schema :only [describe]]
+        [schema.core :only [defschema optional-key enum]])
+  (:import [java.util UUID]))
+
+(defschema AppIdList
+  {:apps (describe [String] "A List of app identifiers")})
+
+(defschema UserPermission
+  {:user       (describe String "The user ID")
+   :permission (describe String "The permission level assigned to the user")})
```
Comment (ianmcorvidae) [inline services/apps/src/apps/routes/domain/permission.clj]: It'd be better if this was an `s/enum` of the available permissions, if that's something we can expect to be able to know in advance.

File: services/apps/src/apps/routes/domain/permission.clj [hunk 2]
```diff
@@ -0,0 +1,18 @@
+(ns apps.routes.domain.permission
+  (:use [common-swagger-api.schema :only [describe]]
+        [schema.core :only [defschema optional-key enum]])
+  (:import [java.util UUID]))
+
+(defschema AppIdList
+  {:apps (describe [String] "A List of app identifiers")})
```
Comment (ianmcorvidae) [inline services/apps/src/apps/routes/domain/permission.clj]: Can this be a UUID instead? Not sure if we need to worry about agave here.

File: services/apps/src/apps/routes/domain/permission.clj [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+(defschema UserPermission
+  {:user       (describe String "The user ID")
```
Comment (ianmcorvidae) [inline services/apps/src/apps/routes/domain/permission.clj]: I'd have this be `NonBlankString` from `common-swagger-api.schema` instead (same for other strings that shouldn't be blank), just for some extra validation/documentation.

File: services/apps/src/apps/routes/domain/permission.clj (hunk 1 above)
Comment (dennisroberts71) [inline services/apps/src/apps/routes/domain/permission.clj]: I considered making it an `enum`, but Grouper supports arbitrary action names. That said, we are sticking to `read`, `write`, `own`, and the empty string right now. I suppose we could make it an enum and change it later if necessary.

File: services/apps/src/apps/routes/domain/permission.clj (hunk 2 above)
Comment (dennisroberts71) [inline services/apps/src/apps/routes/domain/permission.clj]: We do have to worry about agave here. Agave app IDs won't be UUIDs. We're not going to be supporting permission listing for apps immediately, but we will be supporting them.

File: services/apps/src/apps/routes/domain/permission.clj (hunk 1 above)
Comment (ianmcorvidae) [inline services/apps/src/apps/routes/domain/permission.clj]: We depend on those three being the only available things when calculating precedence anyway, looking lower, so I think it'd make sense to enforce it.

Comment (dennisroberts71) [inline services/apps/src/apps/routes/domain/permission.clj]: Yeah, that's a good point. making the change now.

**Comment by ianmcorvidae** [comment]
Aside from those couple of relatively minor comments, lgtm.

**Pull request merged**